source/agora/test/Base.d

```diff
@@ -1263,7 +1263,7 @@ public class TestNetworkManager : NetworkManager
     ***************************************************************************/
 
     protected override TestBanManager getBanManager (in BanManager.Config conf,
-        Clock clock, cstring data_dir)
+        Clock clock, string data_dir)
```

Comment:
Can you explain what the difference is?

Reply:
cstring is more permissive than string in an interface, so cstring should be used:
1. you can pass a string to a cstring argument
2. you cannot pass a cstring to a string argument

honestly, I cannot remember why I made this change... I think it is because this.ban_banfile_path is defined as const string and if you have a cstring then you cannot assign it to const string.

Reply:
I think changing cstring to const cstring in the parameters would have worked too

Reply:
What @ferencdg . I also questioned the change originally, but it is necessary for now. Perhaps later we could just switch the BanManager to use the DB and pass the DB instance, as we already changed for other classes.